### PR TITLE
V2.7.8 work in progress

### PR DESCRIPTION
**Pull request by stieg**
Merge of Firmware release 2.7.8 back into master.

### CODE REVIEW

File: src/logging/printk.c [hunk 1]
```diff
@@ -70,6 +70,12 @@ int writek_int(int value){
 	return write_to_log_buff(buf);
 }
 
+int writek_float(float value){
+	char buf[20];
```
Comment (brentpicasso) [inline src/logging/printk.c]: is this enough for a floating point number?

Comment (stieg) [inline src/logging/printk.c]: Yes.  Floating point has up to 7 digits, plut sign and period and exp values in rare cases.

File: src/tracks/tracks.c [hunk 2]
```diff
@@ -107,7 +107,8 @@ int isStartPointValid(const Track *t) {
    return isValidPoint(&p);
 }
 
-GeoPoint getSectorGeoPointAtIndex(const Track *t, const int index) {
+GeoPoint getSectorGeoPointAtIndex(const Track *t, int index) {
+   if (index < 0) index = 0;
    const int max = isStage(t) ? STAGE_SECTOR_COUNT : CIRCUIT_SECTOR_COUNT;
```
Comment (brentpicasso) [inline src/tracks/tracks.c]: this looks like it might've fixed a crashing bug (checking if < 0) . good catch

Comment (stieg) [inline src/tracks/tracks.c]: actually... no.  But it may have caused the sector stuff to break in a stage scenario.

Comment (stieg) [inline src/tracks/tracks.c]: If you look where the sector array starts on both stage and circuit, you can see the finish point is always at the -1 position to the sector array.

Comment (stieg) [inline src/tracks/tracks.c]: So on a circuit track there would be no problem, but on a stage track the sector logic might possibly not work.  Anyway... its good to have.  I'm talking too much and will shut up startiiiiiinnnnngggg now.

**Pull request merged**